lib/blemulator/internal/bridge/platform_to_dart_bridge.dart

```diff
@@ -3,13 +3,57 @@ part of internal;
 class PlatformToDartBridge {
   SimulationManager _manager;
   MethodChannel _platformToDartChannel;
+  Map<String, CancelableOperation> pendingTransactions = HashMap();
 
   PlatformToDartBridge(this._manager) {
     _platformToDartChannel = new MethodChannel(ChannelName.platformToDart);
     _platformToDartChannel.setMethodCallHandler(_handleCall);
   }
 
-  Future<dynamic> _handleCall(MethodCall call) {
+  Future<dynamic> _handleCall(MethodCall call) async {
+    String transactionId = call.arguments[SimulationArgumentName.transactionId];
+    Future<dynamic> platformCallResultFuture = _dispatchPlatformCall(call);
+
+    if (transactionId != null) {
+      return _createAndSaveCancelablePlatformCall(
+          platformCallResultFuture, transactionId);
+    } else {
+      return platformCallResultFuture;
+    }
+  }
+
+  Future<dynamic> _createAndSaveCancelablePlatformCall(
+    Future<dynamic> platformCallResult,
+    String transactionId,
+  ) {
+    CancelableOperation operation =
+        CancelableOperation.fromFuture(platformCallResult);
+    pendingTransactions.putIfAbsent(transactionId, () => operation);
```

Comment:
And what if it's not absent?

Reply:
I added cancelling & removing pending transactions if they already existed.